src/Compilers/CSharp/Portable/Compilation/CSharpDiagnosticFilter.cs

```diff
@@ -189,6 +192,13 @@ internal static class CSharpDiagnosticFilter
                     specifiedWarnAsErrorMinus = true;
                 }
             }
+            else if (errorCode == (int)ErrorCode.WRN_Experimental)
```

Comment:
@jaredpar @RikkiGibson  This approach should avoid any risk of impact on semantics (we delay the special handling until we're ready to report). 
We could make this logic conditional on `/warnaserror:experimental`, but that still only adds complexity with little value (doesn't avoid the special knowledge of this error code) so I prefer the unconditional approach.